pkg/registry/servicecatalog/binding/storage.go

```diff
@@ -112,45 +110,22 @@ func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, bool, error) {
 
 // NewStorage creates a new rest.Storage responsible for accessing ServiceBinding
 // resources
-func NewStorage(opts server.Options) (rest.Storage, rest.Storage, error) {
-	prefix := "/" + opts.ResourcePrefix()
-
-	storageInterface, dFunc := opts.GetStorage(
-		&servicecatalog.ServiceBinding{},
-		prefix,
-		bindingRESTStrategies,
-		NewList,
-		nil,
-		storage.NoTriggerPublisher,
-	)
-
+func NewStorage(optsGetter generic.RESTOptionsGetter) (serviceBindings, serviceBindingStatus rest.Storage, err error) {
 	store := registry.Store{
-		NewFunc: EmptyObject,
-		// NewListFunc returns an object capable of storing results of an etcd list.
-		NewListFunc: NewList,
-		KeyRootFunc: opts.KeyRootFunc(),
-		KeyFunc:     opts.KeyFunc(true),
-		// Retrieve the name field of the resource.
-		ObjectNameFunc: func(obj runtime.Object) (string, error) {
```

Comment:
Why isn't ObjectNameFunc being set anymore? 

This is getting confusing for me to review... Can you split this up into multiple PRs please?

Reply:
> Why isn't ObjectNameFunc being set anymore?

This is done for free by https://github.com/kubernetes/apiserver/blob/master/pkg/registry/generic/registry/store.go#L1357-L1365 - doesn't need to be set explicitly unless it doesn't use the normal meta name accessor, which in this case it does and hence removed.

> This is getting confusing for me to review... Can you split this up into multiple PRs please?

This PR is composed of 2 commits: the first commit was to simplify storage; the second was asked for by @duglin to deprecate the storage type flag which in the beginning discussion of this PR I recommended to be deprecated, but asked for guidance. I would only see this as a maximum of 2 PRs, one for each of those commits, but you can possibly just as easily review each commit in turn in this PR.